src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  fromFormat.forEach((part, index) => {
+    dateMap[part] = dateParts[index];
+  });
+
+  if (dateMap['YYYY']) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
+  }
```

Comment:
The check for 'YYYY' and subsequent transformation into 'YY' should be conditional based on whether 'YY' is required in the `toFormat`. As it stands, 'YY' is always calculated even if it's not needed.